wp-content/themes/pub/wporg-learn-2020/functions.php

```diff
@@ -290,6 +290,42 @@ function wporg_archive_modify_query( WP_Query $query ) {
 }
 add_action( 'pre_get_posts', 'wporg_archive_modify_query' );
 
+/**
+ * Modify the workshop post type archive query to prioritize workshops in the user's locale.
+ *
+ * @param array    $clauses
+ * @param WP_Query $query
+ *
+ * @return array
+ */
+function wporg_archive_query_prioritize_locale( $clauses, $query ) {
+	if ( ! $query->is_post_type_archive( 'wporg_workshop' ) || is_admin() ) {
+		return $clauses;
+	}
+
+	global $wpdb;
+
+	$locale      = get_locale() ?: 'en_US';
```

Comment:
I'm probably wrong b/c I'm not familiar w/ the locale switching details, but would `get_user_locale()` be better here?

Reply:
In this case I think not, because the Locale Detection plugin filters on the `locale` hook, which gets applied in `get_locale()`, but not necessarily in `get_user_locale()`. I think the user locale option is stored in usermeta and is maybe only used in WP Admin?

Reply:
👍🏻